manifests/backup.pp

```diff
@@ -3,24 +3,23 @@
 # @example
 #   include profile_gitlab::backup
 class profile_gitlab::backup {
+  include ::profile_backup::client
 
-  cron { 'GITLAB DAILY BACKUPS':
-    command => '/opt/gitlab/bin/gitlab-backup create CRON=1',
-    user    => 'root',
-    hour    => 4,
-    minute  => 0,
-    weekday => [
-      '2',
-      '4',
-      '6',
+  profile_backup::client::add_job { 'profile_gitlab':
+    paths             => [
+      '/etc/gitlab/config_backup',
+      '/var/opt/gitlab/backups',
     ],
+    prehook_commands  => '/opt/gitlab/bin/gitlab-backup create CRON=1',
+    posthook_commands => "find /etc/gitlab/config_backup \\( -name '*.tar' \\) -mtime +30 -type f -exec rm -rf {} \\;
+find /var/opt/gitlab/backups \\( -name '*.tar' \\) -mtime +2 -type f -exec rm -rf {} \\;",
   }
 
-  $backup_paths = '/etc/gitlab/config_backup/ /var/opt/gitlab/backups/'
+  # REMOVE OLD BACKUP CRONS
```

Comment:
It's nice to use puppet to do cleanup, but since this is a one-time cleanup I'm wondering if handling this a different way would be better.

Mainly don't want these cleanup lines hanging around into the future where we then wonder what it's even cleaning up (IE someone wondering where these crons were created from in the first place - and why they need cleaning up).

Would manually removing the crons on the hosts before/while flipping them to the new module work? I didn't think we had too many gitlab servers so wasn't thinking it would be too painful, but I could be unaware of something.

Alternatively, if we really needed these in puppet, maybe a comment saying it's removing crons from v0.1.1 of this module? Or making a ticket for ourselves in the future that will clean these out once all gitlab servers are on the new module?

Reply:
My perspective is that if the class previously ensured cron entries that are no longer needed, then the class should remove it. I don't see why that is an issue, but do agree that documenting it better is useful.

In this particular case there are only 2 servers using this class, so I can kind of see it either way.

Reply:
Thanks, with the newer comment it's more clear why those are there and easier for us to remember if we still need it X years from now.